### DISCUSSION TITLE
Reading results for Variables from additional result file

### DISCUSSION THREAD

**Discussion by leszek-z** [DHI/mikeio1d#42] [Q&A] [answered]
Is it possible to read results for variables defined in the MIKE HYDRO? They are stored in an additional result file.
![image](https://user-images.githubusercontent.com/111494126/216363844-da64f4f5-2edd-40eb-9919-d7fc39d63cc7.png)

**Comment by gedaskir** [comment, marked as answer]
For doing this you could try to use a development implementation of mikeio1d from **feature/result_network_auto_completion** branch.  You can install a feature branch using:

```cmd
pip install https://github.com/DHI/mikeio1d/archive/feature/result_network_auto_completion.zip
```

Here I implemented ```QueryDataGlobal```, which can be used as:

```python
from mikeio1d import Res1D
from mikeio1d.query import QueryDataGlobal
res1d = Res1D('MyRes1D.resd')
query_global_data = QueryDataGlobal('Variable:MyVariable')
df = res1d.read(query_global_data)
```

If I remember correctly, the sensor quantities/variables are considered global data items.
Additionally, that branch has auto-completion functionality for Jupyter etc., so now one could add a query using:

```python
from mikeio1d import Res1D
res1d = Res1D('MyRes1D.resd')
res1d.global_data.VariableMyVariable.add()
df = res1d.read()
```

**Reply by gedaskir** [reply]
Now this functionality is made available as part of mikeio1d 0.3.
